cli/tests/unit/io_test.ts

```diff
@@ -71,3 +71,199 @@ unitTest({ perms: { write: true } }, async function copyBufferToFile() {
+      return Deno.writeAll(writer, chunk);
+    },
+  });
+
+  const n = await Deno.copy(reader, writableStream);
+
+  // Since we provided a WritableStream to Deno.copy
+  // Deno.copy should "close" the writer to release the lock, and we can create a new one later
+  // The following should NOT fail with: Cannot create a writer for a locked stream
+  Deno.toWriter(writableStream).close();
+
+  assertEquals(n, xBytes.length);
+  assertEquals(writer.length, xBytes.length);
+});
+
+unitTest(async function copyCheckCloseExternalWritableStreamWriter() {
+  const xBytes = repeat("b", DEFAULT_BUF_SIZE);
+  const reader = new Deno.Buffer(xBytes.buffer as ArrayBuffer);
+  const writer = new Deno.Buffer();
+
+  const writableStream = new WritableStream({
+    write(chunk): Promise<void> {
+      return Deno.writeAll(writer, chunk);
+    },
+  });
+
+  const streamWriter = Deno.toWriter(writableStream);
+  const n = await Deno.copy(reader, streamWriter);
+
+  // Since we provided a WritableStreamIOWriter to Deno.copy
+  // Deno.copy should NOT "close" the writer to release the lock.
+  // The following should fail with: Cannot create a writer for a locked stream
+  assertThrows(
+    () => Deno.toWriter(writableStream),
+    TypeError,
+    "Cannot create a writer for a locked stream"
+  );
+
+  streamWriter.close();
+
+  // Now I can get a new writer
+  Deno.toWriter(writableStream).close();
+
+  assertEquals(n, xBytes.length);
+  assertEquals(writer.length, xBytes.length);
+});
+
+unitTest(async function toWriterCheck() {
+  const written: string[] = [];
+  const chunks: string[] = ["hello", "deno", "land"];
+  const writableStream = new WritableStream({
+    write(chunk): void {
+      const decoder = new TextDecoder();
+      written.push(decoder.decode(chunk));
+    },
+  });
+
+  const encoder = new TextEncoder();
+  const writer = Deno.toWriter(writableStream);
+
+  for (const chunk of chunks) {
+    await writer.write(encoder.encode(chunk));
+  }
+
+  assertEquals(written, chunks);
+
+  // Should have method close
+  writer.close();
+});
+
+unitTest(async function toReaderCheck() {
+  const chunks: string[] = ["hello", "deno", "land"];
+  const expected = chunks.slice();
+  const readChunks: Uint8Array[] = [];
+  const readableStream = new ReadableStream({
+    pull(controller): void {
+      const encoder = new TextEncoder();
+      const chunk = chunks.shift();
+      if (!chunk) return controller.close();
+      controller.enqueue(encoder.encode(chunk));
+    },
+  });
+
+  const decoder = new TextDecoder();
+  const reader = Deno.toReader(readableStream);
+
+  let i = 0;
+
+  while (true) {
+    const b = new Uint8Array(1024);
+    const n = await reader.read(b);
+
+    if (n === null) break;
+
+    readChunks.push(b.subarray(0, n));
+    assert(i < expected.length);
+
+    i++;
+  }
+
+  assertEquals(
+    expected,
+    readChunks.map((chunk) => decoder.decode(chunk))
+  );
+});
+
+unitTest(async function toReaderBigChunksCheck() {
```

Comment:
This test will fail until I implement a buffer mechanism in `RedableStreamIOReader` to buffer exceeding bytes. Since chunk passed to `controller.enqueue` can be greater than `bufSize`.

The buffer mechanism won't be needed for `ReadableStreams` using `ReadableByteStreamController`, since we can pass our own view if we create the reader with `{ mode: 'byob' }`.

(We don't currently support `byob` though)